hips/hip-0019.md

```diff
@@ -0,0 +1,120 @@
+---
+hip: "0019"
+title: "Adoption of CUE for Values Processing and Validation in Helm"
+authors: ["Jiří Altman <jiri.altman@usu.com>", "Jan von Löwenstein <jan.von.loewenstein@sap.com>"]
+created: "2024-08-07"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+This HIP proposes the integration of CUE (Configuration Unification Engine) as the primary tool for processing and validating Helm values files, replacing the existing JSON Schema support. As a superset of YAML, JSON, and JSON Schema, CUE extends existing capabilities while maintaining complete backward compatibility. This ensures that all current Helm charts and configurations remain fully supported, with the added benefits of enhanced validation, improved defaulting mechanisms, and easier schema management.
+
+## Motivation
+
+The current JSON Schema implementation for validating Helm values files has several notable limitations:
+
+1. **Outdated Implementation**: The existing JSON Schema support in Helm is based on Draft-07 and has not been actively maintained. It supports only a subset of the specification and contains non-standard issues, leading to inconsistent behavior and limited validation capabilities.
```

Comment:
minor note: https://github.com/helm/helm/pull/13283 would mean Helm can support newer json schema versions

(if that PR lands, we would want to remove this point I think)